##### Angelica 
##### DSCI_V 100 008
## Project Planning Stage (Individual)


Loading packages and setting a seed for the entire notebook:

In [1]:
# attaching packages
library(tidyverse)
library(tidymodels)

# setting seed
set.seed(160)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

Reading in both the players and sessions datasets:

In [2]:
# reading in players.csv dataset
players_data <- read_csv("players.csv") 
# players_data - to ensure data frame is loaded correctly

# previewing the first 10 rows to save up notebook lines 
head(players_data, 10) 

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1  Pro         TRUE    
2  Veteran     TRUE    
3  Veteran    FALSE    
4  Amateur     TRUE    
5  Regular     TRUE    
6  Amateur     TRUE    
7  Regular     TRUE    
8  Amateur    FALSE    
9  Amateur     TRUE    
10 Veteran     TRUE    
   hashedEmail                                                     
1  f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2  f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3  b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
4  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
5  7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e
6  f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977
7  8e594b8953193b26f498db95a508b03c6fe1c24bb5251d392c18a0da9a722807
8  1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd
9  8b71f4d66a38389b7528bb38ba6eb71157733df7d1740371852a797ae97d82d1
10 bbe2d83de678f519c4b3daa7265e683b4fe2d814077f9094afd11d8f217039ec
   played_hours name      gender Age
1  30.3         Morgan    Male    9 
2   3.8         Christian Male   17 
3   0.0         Blake     Male   17 
4   0.7         Flora     Female 21 
5   0.1         Kylie     Male   21 
6   0.0         Adrian    Female 17 
7   0.0         Luna      Female 19 
8   0.0         Emerson   Male   21 
9   0.1         Natalie   Male   47 
10  0.0         Nyla      Female 22

In [3]:
# reading in sessions.csv dataset
sessions_data <- read_csv("sessions.csv")
# sessions_data - to ensure data frame is loaded in correctly

# previewing the first 10 rows to save up notebook lines 
head(sessions_data, n = 10)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2  36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3  f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5  36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6  bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7  fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8  ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9  96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
   start_time       end_time         original_start_time original_end_time
1  30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2  17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3  25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4  25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5  25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6  23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12      
7  15/04/2024 07:12 15/04/2024 07:21 1.71317e+12         1.71317e+12      
8  21/09/2024 02:13 21/09/2024 02:30 1.72688e+12         1.72689e+12      
9  21/06/2024 02:31 21/06/2024 02:49 1.71894e+12         1.71894e+12      
10 16/05/2024 05:13 16/05/2024 05:52 1.71584e+12         1.71584e+12

## (1) Data Description

### For players.csv dataset:
Notes:
- total number of variables/columns is quantified by visual inspection of the players_data tibble message - later confirmed using summary()
- total number of observations/rows is quantified by visual inspection of the players_data tibble message - later confirmed using summary()

#### General overview of the data
- Total number of observations (rows): 196
- Total number of variables (columns): 7
##### Table 1: Further information about variables

| Variable name    | Variable data type   | What do the variables mean                                                      | 
|:-----------------| --------------- | ------------------------------------------------------------------------------- | 
| experience       | chr (character) | describes experience / skill level     |
| subscribe        | lgl (logical)   | whether players are subscribed to or not to a game-related newsletter              |
| hashedEmail      | chr (character) | an anonymized unique identifier generated by hashing the user's email address   |
| played_hours     | dbl (double)    | total time played (hours) for each unique player (identified by hashedEmail)    |
| name             | chr (character) | player’s display name, randomly generated and chosen by the player              |
| gender           | chr (character) | gender of the player                                                            |
| Age              | dbl (double)    | age of the player (years)                                                       |

#### Summary Statistics

Summary statistics will first be computed, after which the results will be presented in a structured table format.

A general summary computation of all the variables: 

In [4]:
summary(players_data)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

In [54]:
# finding NA entries in all variables
missing_entries <- players_data |>
                        summarize(across(experience:Age, ~sum(is.na(.))))
missing_entries

experience subscribe hashedEmail played_hours name gender Age
1 0          0         0           0            0    0      2

Notes:
- The age variable contains two missing entries (NA's)
    - NA's need to be removed when further exploring summary statistics for the age variable
 - The hashedEmail and name variables contain entirely unique values, with each entry corresponding to a separate individual in the dataset. Since they serve only as anonymous identifiers, they do not offer categorical groupings or numeric data suitable for summary statistics
 - means can't be computed for categorial variables

Using the general summary to further explore individual variables:

In [39]:
# experience variable summary statistics
experience_summary <- players_data |> 
                            group_by(experience) |>
                            summarize(count = n()) |> # summarizes frequency counts of each category
                            mutate(percentage = round((count / sum(count)) * 100, 2)) |> # expresses each category count as a percentage to the total count
                            arrange(desc(count))
experience_summary

experience count percentage
1 Amateur    63    32.14     
2 Veteran    48    24.49     
3 Regular    36    18.37     
4 Beginner   35    17.86     
5 Pro        14     7.14

In [56]:
# subscribe variable summary statistics
subscribe_summary <- players_data |> 
                            group_by(subscribe) |>
                            summarize(count = n()) |> # summarizes frequency counts of each category
                            mutate(percentage = round((count / sum(count)) * 100, 2))  # expresses each category count as a percentage to the total count
subscribe_summary

subscribe count percentage
1 FALSE      52   26.53     
2  TRUE     144   73.47

In [47]:
# played_hours summary statistics 
played_hours_summary <- players_data |>
  summarize(
    min_hours = min(played_hours, na.rm = TRUE), # finding the min number of hours played
    max_hours = max(played_hours, na.rm = TRUE), # finding the max number of hours played
    mean_hours = round(mean(played_hours, na.rm = TRUE), 2), # finding the mean of total hours played
  )
played_hours_summary

min_hours max_hours mean_hours
1 0         223.1     5.85

In [55]:
# gender summary statistics
gender_summary <- players_data |>
                    group_by(gender) |>
                    summarize(count = n()) |> # summarizes frequency counts of each category
                    mutate(percentage = round((count / sum(count)) * 100, 2)) |> # expresses each category count as a percentage to the total count
                    arrange(desc(count))
gender_summary

gender            count percentage
1 Male              124   63.27     
2 Female             37   18.88     
3 Non-binary         15    7.65     
4 Prefer not to say  11    5.61     
5 Two-Spirited        6    3.06     
6 Agender             2    1.02     
7 Other               1    0.51

In [48]:
# Age summary statistics
Age_summary_minMaxMean <- players_data |>
             summarize(
                min_Age = min(Age, na.rm = TRUE), # finding the min number for age
                max_Age = max(Age, na.rm = TRUE), # finding the max number of age
                mean_Age = round(mean(Age, na.rm = TRUE), 2), # finding the mean of all ages
              )   
Age_summary_minMaxMean

# finding the top 3 ages counts
Age_summary_top3_counts <- players_data |>
                                    group_by(Age) |>
                                    summarize(count = n()) |> 
                                    mutate(percentage = round((count / sum(count)) * 100, 2)) |>
                                    arrange(desc(count)) |>
                                    slice_head(n = 3)
Age_summary_top3_counts

min_Age max_Age mean_Age
1 9       58      21.14

Age count percentage
1 17  73    37.24     
2 21  18     9.18     
3 22  15     7.65

#### Brief data interpretation

##### Table 2: Structured summary statistics for all variables

| Variable name | Measurement Type  | Mean              | Other Summaries                                                                                                      | Missing entries (NA's)   |
| ------------- | :----------------:| :---------------: | :------------------------------------------------------------------------------------------------------------------: | :----------------------: |
| played_hours  | numerical         |  5.85             | Max. number of hours recorded for a player:<br>  223.10 <br><br>  Min. number of hours recorded for a player: <br> 0 | 0                        |
| Age           | numerical         |  21.14            | Top 3 most frequent ages by percentage:<br>  17 (37.24 %)<br>    21 (9.18 %)<br>    22 (7.65%)                       | 2                        | 
| experience    | categorical       |       -           | Categories (5):<br> Amateur, Veteran, Regular, Beginner, Pro <br><br> Top 3 most frequent experience levels by percentage:<br> Amateur (32.14 %)<br> Veteran (24.49 %)<br> Regular (18.37 %)                                                                                                                                  | 0                        |
| subscribe     | categorical       |       -           | Categories (logical values):<br> TRUE, FALSE<br><br>   Percentages:<br> TRUE (73.47 %)<br> FALSE (26.53 %)           | 0                        |
| gender        | categorical       |       -           | Categories (7): <br> Male, Female, Non-binary, Prefer not to say, Two-Spirited, Agender, Other<br><br> Top 3 most commonly reported genders by percentage: <br> Male (63.27 %) <br>  Female (18.88 %) <br>  Non-binary (7.65 %)                                                                                              | 0                         |
| hashedEmail   | -                 |       -           |        -                                                                                                             | 0                         |
| name          | -                 |       -           |        -                                                                                                               | 0                       |